core/orchestrator.go

```diff
@@ -628,6 +628,9 @@ func (n *LivepeerNode) transcodeSegmentLoop(md *SegTranscodingMetadata, segChan
 				// timeout; clean up goroutine here
 				os.EndSession()
 				los.EndSession()
+				if n.TranscoderManager != nil {
```

Comment:
Let's add a comment that this check is to avoid a nil pointer error when TranscoderManager is garbage collected while this go routine is still running.